### PR TITLE
Process data into h5 from raw ROOT files

### PR DESCRIPTION
**Pull request by jmduarte**
- Process data into h5 from raw ROOT files

### CODE REVIEW

**Comment by billy000400** [comment]
The generated h5 files passed the H5Data.generate_data() method, so the subgroups are correctly set.

Modifications:
1. Added the path of src/ to sys.path for absolute import (otherwise it cannot find the src module in subdir).
2. Updated requirements.txt to run model.train_model.py

Didn't make it t to training due to the error below:
Scripts were run in a container built upon jmduarte/hbb_interaction_network.
<br class="Apple-interchange-newline">
> setGPU: Setting GPU to: 0
> val data: 0
> train data: 0
> Traceback (most recent call last):
>   File "/cookiecutter-bench/hbb_interaction_network/src/models/train_model.py", line 455, in <module>
>     main(args)
>   File "/cookiecutter-bench/hbb_interaction_network/src/models/train_model.py", line 157, in main
>     gnn = GraphNet(
>   File "/cookiecutter-bench/hbb_interaction_network/src/models/models.py", line 35, in __init__
>     self.assign_matrices()
>   File "/cookiecutter-bench/hbb_interaction_network/src/models/models.py", line 73, in assign_matrices
>     self.Rr = (self.Rr).cuda()
>   File "/opt/conda/lib/python3.9/site-packages/torch/cuda/__init__.py", line 211, in _lazy_init
>     raise AssertionError("Torch not compiled with CUDA enabled")
> AssertionError: Torch not compiled with CUDA enabled

Will inspect tomorrow

**Comment by jmduarte** [comment]
@billy000400 i think the `requirements.txt` file installs this project as a package, so you shouldn't need to add it to `sys.path`

if not, you should do 
```bash
pip install -e .
```
inside the project directory.

**Comment by jmduarte** [comment]
@yorkiva @billy000400 @poem2018 what do you think? is it ok to merge or are there issues with the files?

**Comment by poem2018** [comment]
I may need to adjust some packages import to fit the code organization in repo, hmm it should be fast, I will finish it today.

Ruike
________________________________

**Comment by poem2018** [comment]
Hi Javier,

I finished modifying the import part and tested on thetaGPU, my part should be ok with repo structure now. I put the random split training code in src/models folder, and the onnx and PyTorch related code in src/utils folder. All the saved models will be saved under models/trained_models folder and notebook will be put under ./notebook folder.

Would you mind take a brief look and see if there is anything need modified?


Best,
Ruike
________________________________

**Comment by billy000400** [comment]
The make_dataset.py works flawlessly. I started with a clean conda env and had no problems.

Training didn't work because torch was not compiled with CUDA enabled, and I think the models.py does not support training on  cpu. However, they are not due to the data generating/processing scripts (so unrelated to this pull). I'm working on solving those problems.

File: src/data/definitions.yml [hunk 1]
```diff
@@ -0,0 +1,186 @@
+tree_name: "deepntuplizer/tree"
+n_feature_sets: 4
+batch_size: 100000
+# zero-pad up to 1 jet, 100 pf cands, 60 tracks, 5 svs
+nobj_0: 1
```
Comment (billy000400) [inline src/data/definitions.yml:5]: Could we do nobj_jet, nobj_pf, etc. to make the definitions more readable?

File: src/data/download_raw.py [hunk 2]
```diff
@@ -10,8 +10,11 @@
 data_raw_dir.mkdir(parents=True, exist_ok=True)
```
Comment (billy000400) [inline src/data/download_raw.py]: My original purpose of this script was similar with make_dataset.py except make_dataset.py changes the batch size to 100,000. We could delete this file.

File: src/models/train_model.py [hunk 3]
```diff
@@ -4,86 +4,41 @@
...
-test_path = "dataset/test/"
-train_path = "dataset/train/"
-NBINS = 40  # number of bins for loss function
-MMAX = 200.0  # max value
-MMIN = 40.0  # min value
-
-N = 60  # number of charged particles
-N_sv = 5  # number of SVs
-n_targets = 2  # number of classes
-device = "cpu"
```
Comment (billy000400) [inline src/models/train_model.py:28]: This device variable is not used. Moreover, model.py used .cuda() many times, so we need to add a cpu option if time permits

**Review by billy000400 — APPROVED** [review]

**Comment by jmduarte** [comment]
> Hi Javier, I finished modifying the import part and tested on thetaGPU, my part should be ok with repo structure now. I put the random split training code in src/models folder, and the onnx and PyTorch related code in src/utils folder. All the saved models will be saved under models/trained_models folder and notebook will be put under ./notebook folder. Would you mind take a brief look and see if there is anything need modified? Best, Ruike

@poem2018 can you point me to where our code is? it may be better to make a new PR from a new branch or fork with your code

**Comment by poem2018** [comment]
Got it, I just made a PR from the portability branch.

Ruike
________________________________

**Pull request merged**